monai/deploy/operators/monai_bundle_inference_operator.py

```diff
@@ -701,7 +826,50 @@ def _receive_input(self, name: str, op_input, context):
             logging.debug(f"Shape of the converted input image: {value.shape}")
             logging.debug(f"Metadata of the converted input image: {metadata}")
         elif isinstance(value, np.ndarray):
+            # Keep numpy array as-is when possible and set metadata so downstream transforms handle channels.
+            # Use bundle metadata to infer expected number of channels and adjust conservatively.
+            ndims = value.ndim
+            expected_channels = None
+            try:
+                in_meta = self._inputs.get(name, {})
```

Comment:
What's the logic here? self._inputs are the name of the HSDK operator input ports, not the actual data that will be sent to the input port. Expecting num_channels is not the correct logic.